### PR TITLE
Added header to allow dynamic ttl per request when using greedy cache

### PR DESCRIPTION
**Pull request by Neirda24**
Fix https://github.com/Kevinrob/guzzle-cache-middleware/issues/98

### CODE REVIEW

File: src/Strategy/GreedyCacheStrategy.php [hunk 1]
```diff
@@ -91,7 +93,15 @@ protected function getCacheObject(RequestInterface $request, ResponseInterface $
         }
 
         $response = $response->withoutHeader('Etag')->withoutHeader('Last-Modified');
-        return new CacheEntry($request, $response, new \DateTime(sprintf('+%d seconds', $this->ttl)));
+
+        $ttl = $this->defaultTtl;
+        if ($request->hasHeader(self::HEADER_TTL)) {
+            $ttlHeaderValues = $request->getHeader(self::HEADER_TTL);
+            $ttl = (int)reset($ttlHeaderValues);
+            unset($ttlHeaderValues);
```
Comment (Kevinrob) [inline src/Strategy/GreedyCacheStrategy.php]: Why do you do this `unset`?

**Comment by Kevinrob** [comment]
Thanks for the PR!
Can you add some tests about this feature? If you don't have time, I can do it myself.

File: src/Strategy/GreedyCacheStrategy.php (hunk 1 above)
Comment (Neirda24) [reply src/Strategy/GreedyCacheStrategy.php]: it was just to not keep a variable that has no purposes. I would prefer to not create it but reset takes only variables. Do you prefer I remove this line @Kevinrob ?

**Comment by Neirda24** [comment]
@Kevinrob : I'll try to add some tests. I'll let you know ;)

**Comment by Neirda24** [comment]
@Kevinrob : In fact I don't see how to write this tests as no other tests are ttl oriented.

File: src/Strategy/GreedyCacheStrategy.php (hunk 1 above)
Comment (Kevinrob) [reply src/Strategy/GreedyCacheStrategy.php]: No, I thing that it's okay to do that. Thanks

**Comment by Kevinrob** [comment]
Okay, I will check for adding tests!

**Pull request merged**